### ISSUE TITLE
spotlight state broken (1.4.1), reports "unexpected state"

### ISSUE THREAD

**Issue by JMoVS** [openzfsonosx/zfs#379]
after 7 days of uptime, now spotlight stopped working on O3X 1.4.1, no direct apparent reason.

indexing still working on HFS volumes, just not the ZFS ones
"unexpected indexing state"

neither mdutil `-E`, nor -i off->on nor `rm -rf` the .Spotlight V100 folder help

Some log messages that might be of interest (appeared a couple of times):
`14.09.15 13:36:14,148 mds[63]: (FMW.Warning:1078) event:1 had an arg mismatch. ac:2 am:51`
not many but more than 10

and a heck of a lot of the following messages:
`14.09.15 13:34:11,499 mds[63]: (FMW.Error:387) WE ARE DROPPING FMW EVENTS!`

what I did was installing anaconda and deleting and etc (which somehow involves 150.000 files in the directory). But I did the same to my internal HFS volume and that did survive it.

**Comment by JMoVS** [comment]
here is a spindump: http://cl.ly/d5au

**Comment by JMoVS** [comment]
it's persistent through reboot. I still cannot enable spotlight after having exported and imported the pool again and also having made a reboot

**Comment by JMoVS** [comment]
state is always reported as:
`indexing and search disabled`

**Comment by JMoVS** [comment]
saw that in the log during bootup (they are the datasets):
````
14.09.15 18:25:19,045 mds[63]: (Backup.Error:201) MDSBackupTask:0x0 Undispatched backup task:24582 resolved:0 cid:12:-1 finished:-1
14.09.15 18:25:24,342 mds[63]: (Volume.Normal:2464) volume:0x7fb7c18d7000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/CSFamilienbande/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/CSFamilienbande
14.09.15 18:25:24,791 mds[63]: (Volume.Normal:2464) volume:0x7fb7c191c200 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Documents/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Documents
14.09.15 18:25:31,700 mds[63]: (Volume.Normal:2464) volume:0x7fb7c1217000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Downloads/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Downloads
14.09.15 18:25:32,190 mds[63]: (Volume.Normal:2464) volume:0x7fb7c1053800 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Dropbox/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Dropbox
14.09.15 18:25:32,464 mds[63]: (Volume.Normal:2464) volume:0x7fb7c19ec000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/KundS/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/KundS
14.09.15 18:25:33,056 mds[63]: (Volume.Normal:2464) volume:0x7fb7c19edc00 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Library/Application Support/Firefox/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Library/Application Support/Firefox
14.09.15 18:25:33,924 mds[63]: (Volume.Normal:2464) volume:0x7fb7c1052800 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Library/Application Support/MobileSync/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Library/Application Support/MobileSync
14.09.15 18:25:36,017 mds[63]: (Volume.Normal:2464) volume:0x7fb7c1216000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:(null) occlude:0 /Users/justin/Library/Mail
14.09.15 18:25:36,085 mds[63]: (Volume.Normal:2464) volume:0x7fb7c1a6c000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/MinecraftServerBackups/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/MinecraftServerBackups
14.09.15 18:25:36,158 mds[63]: (Volume.Normal:2464) volume:0x7fb7c134f000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Movies/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Movies
14.09.15 18:25:36,295 mds[63]: (Volume.Normal:2464) volume:0x7fb7c10dc000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Music/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Music
14.09.15 18:25:37,534 mds_stores[340]: (SecureStore.Error:157) Access token 7 changed uid from -1 to 502
14.09.15 18:25:38,201 mds[63]: (Volume.Normal:2464) volume:0x7fb7c1a3ac00 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/Pictures/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/Pictures
14.09.15 18:25:38,354 mds[63]: (Volume.Normal:2464) volume:0x7fb7c193d000 ********** Bootstrapped Creating a default store:0 SpotLoc:(null) SpotVerLoc:/Users/justin/VirtualBox VMs/.Spotlight-V100/Store-V1 occlude:0 /Users/justin/VirtualBox VMs
````

and another mds thing:
````
14.09.15 18:25:47,806 mds[63]: (DiskStore.Normal:2382) 13001 1.028049
````

**Comment by JMoVS** [comment]
can be closed as described in #379 as setting browse=on got it back
